src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -83,41 +126,51 @@ private static JObject ReadProject(string projectJsonPath)
            } 
         }
 
+        private JToken GetFrameworkDependenciesSection(JObject projectJsonRoot, string framework = null)
+        {
+            if(string.IsNullOrWhiteSpace(framework))
```

Comment:
I think there are still a couple things that don't add up for me. One are there any cases where framework will ever be null in this method? If I'm reading the code correctly then I don't think there are any cases.

Second when you are looking for existing dependencies below so you can update add them if they don't exist you aren't taking into account the shared dependencies in the project.json file that aren't under the framework section itself.